### PR TITLE
check hashes for bb files

### PR DESCRIPTION
**Pull request by aaronkl**
*Description of changes:*

replaces  #413

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: syne_tune/blackbox_repository/conversion_scripts/utils.py [hunk 1]
```diff
@@ -66,3 +67,28 @@ def download_file(source: str, destination: str):
...
+    hashes = []
+    for fname in os.listdir(tgt_folder):
+        h = compute_hash(Path(tgt_folder) / fname)
+        hashes.append(h)
+    aggregated_hash = hashlib.sha256()
+    [aggregated_hash.update(h.encode("utf-8")) for h in hashes]
+    return aggregated_hash.hexdigest()
+
+
+def compare_hash(tgt_folder, original_hash):
```
Comment (wistuba) [inline syne_tune/blackbox_repository/conversion_scripts/utils.py:92]: based on the function name, it is not clear that this returns a boolean nor that it computes the hash of the current benchmark and compares it to the expected one.

File: syne_tune/blackbox_repository/repository.py [hunk 2]
```diff
@@ -100,11 +104,28 @@ def load_blackbox(
         return instantiate_yahpo(name, **yahpo_kwargs)
 
     tgt_folder = Path(repository_path) / name
+
+    original_hash = generate_blackbox_recipes[name].hash
```
Comment (wistuba) [inline syne_tune/blackbox_repository/repository.py:108]: expected_hash?

File: syne_tune/blackbox_repository/repository.py [hunk 3]
```diff
@@ -100,11 +104,28 @@ def load_blackbox(
...
         and (tgt_folder / "metadata.json").exists()
         and skip_if_present
     ):
+        if (
+            not ignore_hash
+            and original_hash is not None
+            and not compare_hash(tgt_folder, original_hash)
+        ):
+            logging.warning(
+                f"Files seem to be corrupted (hash missmatch), regenerating it locally and persisting it on S3."
```
Comment (wistuba) [inline syne_tune/blackbox_repository/repository.py:121]: typo missmatch. same below

**Review by wistuba — COMMENTED** [review]
Overall looks okay, left some minor comments

**Review by geoalgo — APPROVED** [review]
Nice!

**Pull request merged**